lib/iris/util.py

```diff
@@ -363,7 +363,8 @@ def array_equal(array1, array2, withnans=False):
 
     def normalise_array(array):
         if not is_lazy_data(array):
-            array = np.asarray(array)
+            if not ma.isMaskedArray(array):
+                array = np.asarray(array)
```

Comment:
Would `np.asanyarray` maybe be a better approach  here ?
I think it would achieve the same thing, and maybe a more standard approach